landlab/components/profiler/trickle_down_profiler.py

```diff
@@ -0,0 +1,392 @@
+# coding: utf8
+# ! /usr/env/python
+"""trickle_down_profiler.py component to create channel profiles."""
+from collections import OrderedDict
+
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+from matplotlib import cm
+
+from landlab.components.profiler.base_profiler import _BaseProfiler
+from landlab.core.utils import as_id_array
+from landlab.utils.flow__distance import calculate_flow__distance
+
+
+class TrickleDownProfiler(_BaseProfiler):
+    """Extract and plot a channel profile from a node ID downstream.
```

Comment:
This is a really informative docstring. Could be used as a model for what a good component docstring looks like. :)

Reply:
Thanks, its basically the same as the ChannelProfiler docstring (which was improved substantially based on @SiccarPoint and @nathanlyons comments)